components/shapediver/parameter/ParameterSelectionComponent.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, Group, Text } from "@mantine/core";
+import React, { useEffect, useState } from "react";
+import ParameterLabelComponent from "./ParameterLabelComponent";
+import { PropsParameter } from "../../../types/components/shapediver/propsParameter";
+import { useParameterComponentCommons } from "../../../hooks/shapediver/parameters/useParameterComponentCommons";
+import { useInteraction } from "shared/hooks/shapediver/viewer/useInteraction";
+import { IconHandFinger, IconInfoCircleFilled } from "@tabler/icons-react";
+import { IInteractionParameterSettings, ISelectionParameterSettings, isInteractionSelectionParameterSettings } from "@shapediver/viewer";
+import { useShapeDiverStoreViewer } from "shared/store/useShapeDiverStoreViewer";
+
+const VIEWPORT_ID = "viewport_1";
+
+/**
+ * Functional component that creates a switch component for a selection parameter.
+ *
+ * @returns
+ */
+export default function ParameterSelectionComponent(props: PropsParameter) {
+
+	// get the parameter API as we might have to overwrite the acceptRejectMode
+	const sessionApi = useShapeDiverStoreViewer(state => { return state.sessions[props.sessionId]; });
+	const parameterApi = sessionApi?.getParameterByName(props.parameterId)[0];
+
+	let acceptRejectMode = props.acceptRejectMode;
+	if(parameterApi && isInteractionSelectionParameterSettings(parameterApi.settings as any)) {
+		const selectionProps = parameterApi.settings?.props as ISelectionParameterSettings;
+		if((selectionProps?.minimumSelection ?? 1) < (selectionProps?.maximumSelection ?? 1)) {
+			acceptRejectMode = true;
+		}
+	}
+
+	const {
+		definition,
```

Comment:
@MajorMeerkatThe3rd you can use `definition` from useParameterComponentCommons instead of calling these hooks again